### PR TITLE
Bug/issues/11

### PR DESCRIPTION
**Pull request by dannynpham**
I created getters to check for length values and units, then used it to check if the value is 0 to default to 1px.

### CODE REVIEW

**Comment by matias-capeletto** [comment]
Thanks for the PR @dannynpham ! Works great, just commented on a small refactoring to avoid duplication. We also already had getters for the value (length) and unit here: https://github.com/Leniolabs/layoutit-grid/blob/main/src/store.js#L16, would you mind using `parseValue(x)` so we only have one place for that logic?

File: src/generateCode.js [hunk 1]
```diff
@@ -30,10 +30,12 @@ export function areaToCSS(area, { parentGrid, useTemplateAreas = true, validTemp
...
-  let css = `  display: grid;
-  grid-template-columns: ${namedTemplateColumns(grid, repeat)};
-  grid-template-rows: ${namedTemplateRows(grid, repeat)};
-  gap: ${grid.row.gap + ' ' + grid.col.gap};` // TODO: cssGridGap(grid)
+  let css = `
+    display: grid;
+    grid-template-columns: ${namedTemplateColumns(grid, repeat)};
+    grid-template-rows: ${namedTemplateRows(grid, repeat)};
+    gap: ${grid.row.gap} ${grid.col.gap};
+  ` // TODO: cssGridGap(grid)
```
Comment (matias-capeletto) [inline src/generateCode.js:38]: I like this change, but please remove the added newline in line 33, and the one at the end of line 37, or the generated code in CodePen will have extra lines there (this code is only used when exporting to CodePen at the moment)

**Review by matias-capeletto — CHANGES_REQUESTED** [review]

File: src/utils.js [hunk 2]
```diff
@@ -263,3 +263,19 @@ function getCaretCharacterOffsetWithin(element) {
...
+  return match[0]
+}
+
+export function getLengthUnit(size) {
+  if (typeof size !== 'string') throw Error('Size should be string')
+
+  const match = size.match(/[^0-9,.]+/)
+  if (!match) throw Error('No length unit matched')
+  return match[0]
+}
```
Comment (matias-capeletto) [inline src/utils.js:281]: We can remove these ones, and use `parseValue` from store.js. We already have these getters there 👍

File: src/components/grid/GridEditor.vue [hunk 3]
```diff
@@ -93,6 +94,22 @@ export function gridComputedStyles() {
...
+    gap = `${gap} ${grid.value.row.gap} `
+  } else {
+    gap = `${gap} 1px `
+  }
+  if (Number(getLengthValue(grid.value.col.gap))) {
+    gap = `${gap} ${grid.value.col.gap}`
+  } else {
+    gap = `${gap} 1px`
+  }
+  return gap
```
[the remark this answers is not in the document]
Comment (dannynpham) [reply src/components/grid/GridEditor.vue]: That's definitely easier to read. Updated.

File: src/generateCode.js (hunk 1 above)
Comment (dannynpham) [reply src/generateCode.js]: Done

File: src/utils.js (hunk 2 above)
Comment (dannynpham) [reply src/utils.js]: Done

File: src/components/grid/GridEditor.vue [hunk 4]
```diff
@@ -3,9 +3,9 @@
     ref="sectionElement"
     :class="{ active: isCurrent, dragging }"
     :style="{
-      'grid-template-rows': grid.row.sizes.join(' '),
-      'grid-template-columns': grid.col.sizes.join(' '),
-      'grid-gap': grid.row.gap + ' ' + grid.col.gap,
+      gridTemplateRows: grid.row.sizes.join(' '),
+      gridTemplateColumns: grid.col.sizes.join(' '),
+      gridGap: gridGap,
```
Comment (centrodph) [inline src/components/grid/GridEditor.vue:8]: `gridGap,`  can work?

Comment (dannynpham) [reply src/components/grid/GridEditor.vue]: Done.

**Review by matias-capeletto — APPROVED** [review]

**Comment by matias-capeletto** [comment]
Merged! Thanks a lot @dannynpham ! 🎉

**Comment by dannynpham** [comment]
You're welcome! I use your app so often when I found out it was open source I felt compelled to help out :)

**Comment by matias-capeletto** [comment]
That is awesome @dannynpham 💯 
Thanks again and stay around 🙌

**Comment by matias-capeletto** [comment]
@dannynpham https://twitter.com/Layoutit/status/1310656798295285760 ❤️

**Pull request merged**